components/gitpod-db/src/typeorm/team-db-impl.ts

```diff
@@ -122,27 +124,30 @@ export class TeamDBImpl implements TeamDB {
 
     public async createTeam(userId: string, name: string): Promise<Team> {
         if (!name) {
-            throw new Error("Team name cannot be empty");
+            throw new ResponseError(ErrorCodes.BAD_REQUEST, "Team name cannot be empty");
```

Comment:
As a general point about all of these changes, should we throwing `ResponseErrors` from the `gitpod-db` component? We don't throw these errors anywhere else in `gitpod-db`.

Reply:
In general, I'd also be reluctant but we'd need to define a new set of errors, catch these and then re-throw. Given how intertwined the db model is with server, I see somewhat limited harm in doing this here tho. It still throws, just more specific.

Reply:
But happy to go another route if we think it's worth it.

Reply:
I think Andrew's comment is on-point: It feels like this method belongs more into `server`, or even `protocol`.

For (nearly?) all other entities we have those methods in `protocol` (cmp.: [TeamSubscription](https://github.com/gitpod-io/gitpod/blob/5c1074bbfcdfc07829f87c6a7085f3146eb1b3af/components/gitpod-protocol/src/team-subscription-protocol.ts#L26-L30)).

You could still argue whether there shouldn't be a "model-level" error in between. But would still feel better :upside_down_face:

Reply:
So if I understand this correctly, you wan to move this method over to `server` or `protocol`. Once there, do we want to re-map the errors or are we okay to keep them as they are?

Reply:
I'm +1 on this change as I don't think having error translations all over the place helps us.

Reply:
> error translations all over the place helps us.

Valid point.

Let's go with the change as-is, because:
 1. it fixes a concrete pain-point
 2. we don't really have a working counter proposal that is either a) established or b) we see any immediate benefit in

If we resolve 2. at some point in the future, we can always go back and adjust. :slightly_smiling_face: